spec/requests/api/v2/providers/courses_spec.rb

```diff
@@ -151,6 +151,101 @@
     end
   end
 
+  describe 'POST publish' do
+    let(:api_status)   { 200 }
+    let(:api_response) { '{ "result": true }' }
+    let!(:stubbed_manage_courses_api_request) do
+      stub_request(:post, %r{#{Settings.manage_api.base_url}/api/Publish/internal/course/})
+        .to_return(
+          status: api_status,
+          body: api_response
+        )
+    end
+    let(:course) {
+      create(:course,
+             provider: provider,
+             with_site_statuses: [:new],
+             with_enrichments: [:initial_draft])
+    }
+    let(:publish_path) do
+      "/api/v2/providers/#{provider.provider_code}" +
+        "/courses/#{course.course_code}/publish"
+    end
+
+    subject do
+      post publish_path, headers: { 'HTTP_AUTHORIZATION' => credentials }
+      response
+    end
+
+    context 'when unauthenticated' do
+      let(:payload) { { email: 'foo@bar' } }
+
+      it { should have_http_status(:unauthorized) }
+    end
+
+    context 'when user has not accepted terms' do
+      let(:user)         { create(:user, accept_terms_date_utc: nil) }
+      let(:organisation) { create(:organisation, users: [user]) }
+
+      it { should have_http_status(:forbidden) }
+    end
+
+    context 'when unauthorised' do
+      let(:unauthorised_user) { create(:user) }
+      let(:payload)           { { email: unauthorised_user.email } }
+
+      it "raises an error" do
+        expect { subject }.to raise_error Pundit::NotAuthorizedError
+      end
```

Comment:
Can be a one-liner.

Reply:
This makes the test fail, not sure how to make it work:

```
-      it "raises an error" do
-        expect { subject }.to raise_error Pundit::NotAuthorizedError
-      end
+      it { should raise_error Pundit::NotAuthorizedError }
```
failure:
```
$ bundle exec rspec ./spec/requests/api/v2/providers/courses/publish_spec.rb:58                                                                            {1}
Run options: include {:focus=>true, :locations=>{"./spec/requests/api/v2/providers/courses/publish_spec.rb"=>[58]}}
F

Failures:

  1) Publish API v2 POST publish when unauthorised 
     Failure/Error: authorize @course

     Pundit::NotAuthorizedError:
       not allowed to publish? this #<Course id: 1, age_range: nil, course_code: "C1D3", name: "AngelScript", profpost_flag: nil, program_type: "higher_education_programme", qualification: "pgce_with_qts", start_date: nil, study_mode: "full_time", accrediting_provider_id: nil, provider_id: 1, modular: nil, english: nil, maths: nil, science: nil, created_at: "2019-04-25 10:15:29", updated_at: "2019-04-25 10:15:29", changed_at: "2019-04-25 10:15:29">
     # /home/tim/.rvm/gems/ruby-2.6.1/gems/pundit-2.0.1/lib/pundit.rb:221:in `authorize'
     # ./app/controllers/api/v2/courses_controller.rb:49:in `build_course'
     # /home/tim/.rvm/gems/ruby-2.6.1/gems/audited-4.8.0/lib/audited/sweeper.rb:14:in `around'
     # /home/tim/.rvm/gems/ruby-2.6.1/gems/sentry-raven-2.9.0/lib/raven/integrations/rails/controller_transaction.rb:7:in `block in included'
     # /home/tim/.rvm/gems/ruby-2.6.1/gems/audited-4.8.0/lib/audited/sweeper.rb:14:in `around'
     # /home/tim/.rvm/gems/ruby-2.6.1/gems/rack-2.0.7/lib/rack/etag.rb:25:in `call'
     # /home/tim/.rvm/gems/ruby-2.6.1/gems/rack-2.0.7/lib/rack/conditional_get.rb:38:in `call'
     # /home/tim/.rvm/gems/ruby-2.6.1/gems/rack-2.0.7/lib/rack/head.rb:12:in `call'
     # /home/tim/.rvm/gems/ruby-2.6.1/gems/railties-5.2.3/lib/rails/rack/logger.rb:38:in `call_app'
     # /home/tim/.rvm/gems/ruby-2.6.1/gems/railties-5.2.3/lib/rails/rack/logger.rb:26:in `block in call'
     # /home/tim/.rvm/gems/ruby-2.6.1/gems/railties-5.2.3/lib/rails/rack/logger.rb:26:in `call'
     # /home/tim/.rvm/gems/ruby-2.6.1/gems/rack-2.0.7/lib/rack/runtime.rb:22:in `call'
     # /home/tim/.rvm/gems/ruby-2.6.1/gems/rack-2.0.7/lib/rack/sendfile.rb:111:in `call'
     # /home/tim/.rvm/gems/ruby-2.6.1/gems/sentry-raven-2.9.0/lib/raven/integrations/rack.rb:51:in `call'
     # /home/tim/.rvm/gems/ruby-2.6.1/gems/railties-5.2.3/lib/rails/engine.rb:524:in `call'
     # /home/tim/.rvm/gems/ruby-2.6.1/gems/rack-test-1.1.0/lib/rack/mock_session.rb:29:in `request'
     # /home/tim/.rvm/gems/ruby-2.6.1/gems/rack-test-1.1.0/lib/rack/test.rb:266:in `process_request'
     # /home/tim/.rvm/gems/ruby-2.6.1/gems/rack-test-1.1.0/lib/rack/test.rb:119:in `request'
     # /home/tim/.rvm/gems/ruby-2.6.1/gems/rails-controller-testing-1.0.4/lib/rails/controller/testing/integration.rb:13:in `block (2 levels) in <module:Integration>'
     # ./spec/requests/api/v2/providers/courses/publish_spec.rb:37:in `block (3 levels) in <top (required)>'
     # ./spec/requests/api/v2/providers/courses/publish_spec.rb:58:in `block (4 levels) in <top (required)>'
     # ./spec/rails_helper.rb:64:in `block (3 levels) in <top (required)>'
     # /home/tim/.rvm/gems/ruby-2.6.1/gems/database_cleaner-1.7.0/lib/database_cleaner/generic/base.rb:16:in `cleaning'
     # /home/tim/.rvm/gems/ruby-2.6.1/gems/database_cleaner-1.7.0/lib/database_cleaner/base.rb:100:in `cleaning'
     # /home/tim/.rvm/gems/ruby-2.6.1/gems/database_cleaner-1.7.0/lib/database_cleaner/configuration.rb:86:in `block (2 levels) in cleaning'
     # /home/tim/.rvm/gems/ruby-2.6.1/gems/database_cleaner-1.7.0/lib/database_cleaner/configuration.rb:87:in `cleaning'
     # ./spec/rails_helper.rb:63:in `block (2 levels) in <top (required)>'

Finished in 0.80168 seconds (files took 1.45 seconds to load)
1 example, 1 failure

Failed examples:

rspec ./spec/requests/api/v2/providers/courses/publish_spec.rb:58 # Publish API v2 POST publish when unauthorised 

Coverage report generated for RSpec to /home/tim/repo/dfe/manage/backend/coverage. 641 / 807 LOC (79.43%) covered.
```

Reply:
discussed in person, we could come back to this if we have more ideas